Sources/PackageModel/Destination.swift

```diff
@@ -232,28 +233,56 @@ extension Destination {
     public init(fromFile path: AbsolutePath, fileSystem: FileSystem) throws {
         let decoder = JSONDecoder.makeWithDefaults()
         let version = try decoder.decode(path: path, fileSystem: fileSystem, as: VersionInfo.self)
+        
         // Check schema version.
-        guard version.version == 1 else {
+        if version.version == 1 {
+            let destination = try decoder.decode(path: path, fileSystem: fileSystem, as: DestinationInfoV1.self)
+            try self.init(
+                target: destination.target.map{ try Triple($0) },
+                sdk: destination.sdk,
+                binDir: destination.binDir,
+                extraCCFlags: destination.extraCCFlags,
+                extraSwiftCFlags: destination.extraSwiftCFlags,
+                extraCXXFlags: destination.extraCPPFlags
+            )
+        } else if version.version == 2 {
+            let destination = try decoder.decode(path: path, fileSystem: fileSystem, as: DestinationInfoV2.self)
+            let destinationDirectory = path.parentDirectory
+            
+            try self.init(
+                target: destination.target.map{ try Triple($0) },
+                sdk: destination.sdk?.absolutePath(relativeTo: destinationDirectory),
+                binDir: destination.binDir.absolutePath(relativeTo: destinationDirectory),
+                extraCCFlags: destination.extraCCFlags,
+                extraSwiftCFlags: destination.extraSwiftCFlags,
+                // maintaining `destination.extraCPPFlags` naming inconsistency for compatibility.
+                extraCXXFlags: destination.extraCPPFlags
+            )
+        } else {
             throw DestinationError.invalidSchemaVersion
         }
-        let destination = try decoder.decode(path: path, fileSystem: fileSystem, as: DestinationInfo.self)
-        try self.init(
-            target: destination.target.map{ try Triple($0) },
-            sdk: destination.sdk,
-            binDir: destination.binDir,
-            extraCCFlags: destination.extraCCFlags,
-            extraSwiftCFlags: destination.extraSwiftCFlags,
-            // maintaining `destination.extraCPPFlags` naming inconsistency for compatibility.
-            extraCXXFlags: destination.extraCPPFlags
-        )
+    }
+}
+
+fileprivate extension FilePath {
+    /// Converts `FilePath` to `AbsolutePath`, in relation to other `AbsolutePath` if possible.
+    /// - Parameter directory: an absolute path to a directory used to calculate a resulting `AbsolutePath`.
+    /// - Returns: trivially converts `self` to `AbsolutePath` as a string if `self` is an absolute path.
+    /// Otherwise, if `self` is a relative path, returns `self` appended to the `directory` argument.
+    func absolutePath(relativeTo directory: AbsolutePath) throws -> AbsolutePath {
+        if isRelative {
+            return directory.appending(RelativePath(string))
+        } else {
+            return try AbsolutePath(validating: string)
+        }
```

Comment:
iirc the absolute path constructor already takes care of it? @neonichu ?

Reply:
Yah, `init(validating:, relativeTo:)` should do the same thing, I believe